### PR TITLE
RT-23: Static analysis

### PR DESCRIPTION
**Pull request by flavgj**
JIRA LInk: https://gpitbjss.atlassian.net/browse/RT-23
Confluence Link: https://gpitbjss.atlassian.net/wiki/x/XQFfXQ

This will allow for sonar analysis to be triggered by each member of the team from their own machine, and will automatically trigger sonar analysis on any module using the existing `executeUnitTestsWithCoverage` Jenkins function.

### CODE REVIEW

**Comment by flavgj** [comment]
Revoked jenkins sonar token which was in the jenkinsfile.

File: Jenkinsfile [hunk 1]
```diff
@@ -82,4 +82,5 @@ void executeUnitTestsWithCoverage() {
     sh label: 'Running unit tests', script: 'pipenv run unittests-cov'
     sh label: 'Displaying code coverage report', script: 'pipenv run coverage-report'
     sh label: 'Exporting code coverage report', script: 'pipenv run coverage-report-xml'
+    sh label: 'Running SonarQube analysis', script: "sonar-scanner -D${SONAR_HOST} -D${SONAR_TOKEN}""
```
Comment (ghost) [inline Jenkinsfile]: Can you add these two variables to the [pipeline README's Permissions and global variables section](https://github.com/nhsconnect/integration-adaptors/blob/RT-23-static-analysis/pipeline/README.md#permissions-and-global-variables) please?

**Review by ghost — DISMISSED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (ismail-s) [inline Jenkinsfile]: Do these environment variables end up looking like `sonar.blah=asdfkjalkf`? If so, then the `sonar.blah=` parts can be put here.

File: common/README.md [hunk 2]
```diff
@@ -15,3 +15,14 @@ pipenv install -e ./../common
...
 ### Run Tests
 `pipenv run tests` will run all tests.
+
+### Coverage
+`pipenv run unittests-cov` will run all unit tests with coverage enabled. \
+`pipenv run coverage-report-xml` will generate an xml file which can then be submitted for coverage analysis. \
+
+### Analysis
+To use sonarqube analysis you will need to have installed `sonar-scanner`. \
+Ensure sonar-scanner is on your path, and configured for the sonarqube host with appropriate token. \
```
Comment (ismail-s) [inline common/README.md:29]: How do you do this configuration? If there's a webpage that explains this, then can you link to that?

**Review by ismail-s — DISMISSED** [review]

File: mhs/README.md [hunk 3]
```diff
@@ -31,17 +31,20 @@ pipenv install --dev
 ## Running Unit Tests
 `pipenv run unittests` will run all unit tests.
 
-### Coverage
+<<<<<<< HEAD:mhs-reference-implementation/README.md
```
Comment (ismail-s) [inline mhs/README.md]: <<<

File: mhs/README.md [hunk 4]
```diff
@@ -31,17 +31,20 @@ pipenv install --dev
...
 To use sonarqube analysis you will need to have installed `sonar-scanner`. \
-Ensure sonar-scanner is on your path, and configured for the sonarqube host with appropriate token. \
+Ensure sonar-scanner is on your path, and configured for the sonarqube host with appropriate token.
+ (See: [SonarQube](https://gpitbjss.atlassian.net/wiki/x/XQFfXQ))\
 `sonar-scanner` will use `sonar-project.properties` to submit source to sonarqube for analysis. \
 NOTE: Coverage will not show in the analysis unless you have already generated the xml report (as per above.)
 
-### Running Integration Tests
+## Running Integration Tests
+>>>>>>> 618e6863c4f9bdb37b91284f276c578c04cdb437:mhs/README.md
```
Comment (ismail-s) [inline mhs/README.md]: `>>>`

**Review by ismail-s — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]

**Review by ismail-s — APPROVED** [review]

**Pull request merged**